### ISSUE TITLE
Option to invert behavior?

### ISSUE THREAD

**Issue by airdrummingfool** [tjsail33/angular-indeterminate#9]
Hello, and thanks for this great module.

I was wondering if you'd be willing to add a config option to invert the behavior of the indeterminate checkbox. For example, right now when some boxes are checked, clicking the indeterminate box checks all. Many websites (such as Gmail) have the behavior that with some boxes checked, clicking the indeterminate box unchecks all.

The code change to invert the behavior is a small change to line 41:
```
if (enabled.length < values.length) {
```
to:
```
if (enabled.length == 0) {
```

Unfortunately, I don't know enough about angular modules to implement a configuration option. I poked around for a bit but couldn't figure it out from the docs.

**Comment by timcosta** [comment]
Published as v3.1.0, enjoy! Just set `indeterminate-click-behavior="clear"`

**Comment by airdrummingfool** [comment]
Thank you so much! I don't know why I didn't think of using an attribute option.

Unfortunately, right now the  `indeterminate-click-behavior=clear` does not seem to be correct when nothing is selected (i.e. `enabled.length == 0`). Checking the (empty) indeterminate checkbox should select all, but instead it doesn't do anything (because `setValue` is set to `falseValue` instead of `trueValue`). I can work up a PR in a bit to handle that case.
